src/middlewared/middlewared/etc_files/local/nslcd.conf

```diff
@@ -3,50 +3,50 @@
 # $FreeBSD$
 #
 <%
-        ldap = middleware.call_sync('datastore.config', 'directoryservice.LDAP')
+        ldap = middleware.call_sync('ldap.config')
         if ldap:
             certpath = None
-            if ldap['ldap_certificate']:
+            if ldap['certificate']:
                 try:
-                    cert = middleware.call_sync('certificate.query', [('id', '=', ldap['ldap_certificate']['id'])], {'get': True})
+                    cert = middleware.call_sync('certificate.query', [('id', '=', ldap['certificate']['id'])], {'get': True})
                 except IndexError:
                     pass
                 else:
                     certpath = cert['certificate_path']
         else:
             ldap = None
 
-        ldap_enabled = ldap['ldap_enable']
+        ldap_enabled = ldap['enable']
 
-        ldap_uri = "%s://%s" % ("ldaps" if ldap['ldap_ssl'] == "on" else "ldap", ldap['ldap_hostname'])
+        ldap_uri = ["%s://%s" % ("ldaps" if ldap['ssl'] == "on" else "ldap", x) for x in ldap['hostname']]
```

Comment:
According to ldap.update, ssl should be ON, OFF or START_TLS

Is .config returning lowercase? if so we should fix that

Reply:
Ah, that error crept in due to switching from datastore.query to ldap.config. Good catch.